django_cloud_deploy/config.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A module about YAML formated configuration files generation.
+
+The configuration files are expected to capture the state of Django projects. It
+should save the following attributes of a Django project:
+    GCP project id
+    Django project name
+"""
+
+import os
+from typing import Optional
+import yaml
+
+
+class Configuration(object):
+    """A class to encapsulate a single configuration."""
+
+    _HEADER = '# Generated file, do not edit'
+
+    def __init__(self, django_directory_path: str):
+        # If the provided path is not a valid directory path, an error will be
+        # raised.
+        if not os.path.isdir(django_directory_path):
+            raise ValueError('[{}] is not a valid directory path.'.format(
+                django_directory_path))
+        self._config_path = os.path.join(django_directory_path,
+                                         '.config.yaml')
+
+        # It checks if the configuration file exists. If it exists, then the
+        # data from that configuration file will be loaded. Otherwise a new
+        # configuration file will be created by the "save" method.
+        if os.path.exists(self._config_path):
+            with open(self._config_path) as config_file:
+                self._data = yaml.load(config_file)
+        else:
+            self._data = {}
+
+    def set(self, attr: str, value: str):
+        self._data[attr] = value
+
+    def save(self):
+        """Generate the configuration file in yaml format."""
+        yaml_text = '\n'.join([self._HEADER,
+                               yaml.dump(self._data, default_flow_style=False)])
+        with open(self._config_path, 'w') as config_file:
+            config_file.write(yaml_text)
+
+    def get(self, attr: str) -> Optional[str]:
+        """Get the value of the specified attribute.
+
+        Args:
+            attr: The attribute name we want to get value.
+
+        Returns:
+            Value of the specified attribute if it exist. Otherwise return None.
+        """
+        return self._data.get(attr, None)
```

Comment:
You could also write this as
```suggestion
        return self._data.get(attr)
```

But it is OK to be explicit as well. You decide.

Reply:
I removed "None".